### ISSUE TITLE
No clipboard sharing with Kubuntu 22.10 guest

### ISSUE THREAD

**Issue by HautDam** [vmware/open-vm-tools#629]
### Describe the bug

**No clipboard sharing in either direction.**
File drag-n-drop doesn't work, either. 
Folder sharing and screen resizing do work. 

Setup details:
- Win10 host running Workstation 16.2 Player. 
- Fresh install of Kubuntu 22.10 guest.
- open-vm-tools and open-vm-tools-desktop installed and up-to-date.
- I set both isolation.tools.copy 'n paste.disable to false in the vmx file.
- Clipboard sharing does work in my Kubuntu 22.04 VM.

### Reproduction steps

1. Fresh installation of Kubuntu 22.10 from ISO image.
2. Install open-vm-tools.
3. Reboot VM and attempt clipboard sharing between guest and host.

### Expected behavior

Contents of copy buffer should paste in either Linux or Windows.

**Comment by dsouders** [comment]
Thank you for reporting this issue.  I have filed an internal bug to track the problem.

**Comment by johnwvmw** [comment]
This appears to be another variation on the KDE / XDG autostart issues reported in other open-vm-tools issues such as #627.   I have reproduced the problem with Kubuntu 22.10 installed on Workstation 16.2.4.  After logging in on the KDE desktop, the "vmtoolsd -n vmusr" process is not running.

root@Kubuntu-22-10:/home/jwolfe# ps -ef | grep tools
root         591       1  0 14:16 ?        00:00:00 /usr/bin/vmtoolsd
root        1183    1175  0 14:17 pts/1    00:00:00 grep --color=auto tools

The journal logs show that autostart did attempt to run  the the "vmtoolsd -n vmusr" process.

root@Kubuntu-22-10:/home/jwolfe# journalctl -b | grep vmtoolsd
Jan 01 14:17:53 Kubuntu-22-10 dbus-daemon[1079]: [session uid=1000 pid=1079] Activating via systemd: service name='org.a11y.Bus' unit='at-spi-dbus-bus.service' requested by ':1.40' (uid=1000 pid=1534 comm="/usr/bin/vmtoolsd -n vmusr --blockFd 3" label="unconfined")
Jan 01 14:17:53 Kubuntu-22-10 vmtoolsd[1534]: gtk_disable_setlocale() must be called before gtk_init()
Jan 01 14:17:54 Kubuntu-22-10 at-spi-bus-launcher[1600]: dbus-daemon[1600]: Activating service name='org.a11y.atspi.Registry' requested by ':1.0' (uid=1000 pid=1534 comm="/usr/bin/vmtoolsd -n vmusr --blockFd 3" label="unconfined")

Work-around for now is to manually start the the "vmtoolsd -n vmusr" process after logging in to the desktop by running **/usr/bin/vmware-user** from a Konsole window.

root@Kubuntu-22-10:/home/jwolfe# ps -ef | grep tools
root         591       1  0 14:16 ?        00:00:00 /usr/bin/vmtoolsd
jwolfe      1678    1062  1 14:18 pts/3    00:00:00 /usr/bin/vmtoolsd -n vmusr --blockFd 3
root        1689    1175  0 14:18 pts/1    00:00:00 grep --color=auto tools

Copy - paste and drag -n-drop works at this point.

**Comment by HautDam** [comment]
I gave a thumbs up, but probably should have been more explicit: Running vmware-user manually got clipboard- and file-sharing working. Thank you @johnwvmw!

**Comment by ledroide** [comment]
I confirm that running either `vmtoolsd -n vmusr` or `vmware-user` solves the issue for me.
Configuration : 
vmtoolsd -v -> VMware Tools daemon, version 12.1.0.37487 (build-20219665)
uname -a -> Linux cosmic 5.19.0-29-generic #30-Ubuntu SMP PREEMPT_DYNAMIC Wed Jan 4 12:14:09 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
VMwar Workstation Player -> 16.2.5 build-20904516
session type : x11 (not wayland)

**Comment by lamskoy** [comment]
This stuff helped me 
Run and reboot after:
`kwriteconfig5 --file startkderc --group General --key systemdBoot false`

**Comment by polarathene** [comment]
More appropriate workaround is detailed here: https://github.com/vmware/open-vm-tools/issues/568#issuecomment-1577875270

If no one finds an issue with it, then it would be easier to contribute upstream than relying on prior workarounds.

**Comment by lamskoy** [comment]
> More appropriate workaround is detailed here: [#568 (comment)](https://github.com/vmware/open-vm-tools/issues/568#issuecomment-1577875270)
> 
> If no one finds an issue with it, then it would be easier to contribute upstream than relying on prior workarounds.

systemd 254 is available on my ArchLinux installation. Have removed ~/.config/startkderc and rebooted after systemd upgrade. Issue has gone away

**Comment by polarathene** [comment]
> Issue has gone away

Yep, I pointed out that from the upstream bug report I raised, a fix in systemd v254 should resolve it: https://github.com/vmware/open-vm-tools/issues/568#issuecomment-1641191539

**Comment by liangqi** [comment]
My env is VMware Fusion 13.0.2 on macOS 13.5.2, neon(based on Ubuntu) unstable, KDE/Plasma 5.27.80 based on Qt 6.6.0. The workaround "vmtoolsd -n vmusr" works for me.

**Comment by courtarro** [comment]
> vmtoolsd -n vmusr

For the record this does not work for me on a VMWare 17.5, Windows 10 host, KDE Neon guest

**Comment by marbaa** [comment]
VMware player 17.5, Guest Debian 12 KDE, Host Win11. Clipboard mostly doesn't work, open-vm-tools-desktop already installed. This bug is so annoying, probably it is long time existing KDE bug.
